fakebook/src/app/component/newsfeed/newsfeed.component.ts

```diff
@@ -22,23 +22,26 @@ export class NewsfeedComponent implements OnInit {
     ) {}
 
   ngOnInit(): void {
-    this.getUser();
     this.route.params.subscribe((params) => {
-      if(params['id'] !== undefined) {
+      let loadUserFeed = params['id'] !== undefined;
+      this.getPosts(loadUserFeed);
+      if(!loadUserFeed) {
         this.postId = +params['id'];
         if(this.postId !== undefined && !isNaN(this.postId)) {
           this.getPostById();
         }
-      } else {
-        this.getPosts();
-      }
+      } 
     });
   }
 
-  getPosts(): void {
+  getPosts(loadUserFeed : boolean): void {
```

Comment:
what for? you're not even using the parameter.

Reply:
Oops, forgot to finish that in the rush of things. The method should now only load the user feed if it is true.